### PR TITLE
[MOD-2766] Wrote function to check/validate environment names

### PR DESCRIPTION
**Pull request by mattnappo**
Addresses MOD-2766 by writing a function to check/verify environment names.

### CODE REVIEW

**Comment by mwaskom** [comment]
Nice! Would be great to have a couple simple tests; the other code here is tested (somewhat non-obviously) in `test/utils_test.py`

File: test/utils_test.py [hunk 1]
```diff
@@ -29,6 +34,13 @@ def test_object_name():
         check_object_name("foo/bar", "Volume", warn=True)
 
 
+def test_environment_name():
+    assert is_valid_environment_name("AB13/root:env1")
+    assert not is_valid_environment_name("--help")
+    assert not is_valid_environment_name(":env")
+    assert not is_valid_environment_name("/env")
```
Comment (thundergolfer) [inline test/utils_test.py:41]:
```suggestion
    assert not is_valid_environment_name("/env")
    assert not is_valid_environment_name("")
```

**Review by thundergolfer — APPROVED** [review]
Nice 👌

File: test/utils_test.py (hunk 1 above)
Comment (mwaskom) [reply test/utils_test.py:41]: Check also exercise the "not too long" rule with something like `"a" * 65`

**Pull request merged**